packages/template-retail-react-app/app/components/_app-config/index.jsx

```diff
@@ -55,7 +55,9 @@ const AppConfig = ({children, locals = {}}) => {
             redirectURI={`${appOrigin}/callback`}
             proxy={`${appOrigin}${commerceApiConfig.proxyPath}`}
             headers={headers}
-            OCAPISessionsURL={`${appOrigin}/mobify/proxy/ocapi/s/${locals.site?.id}/dw/shop/v22_8/sessions`}
+            OCAPISessionsURL={`${appOrigin}/on/demandware.store/Sites-${
+                locals.site?.id
+            }-Site/${locals.locale?.id?.replace('-', '_')}/SLASSessionHelper-SyncHybridSession`}
```

Comment:
```suggestion
            OCAPISessionsURL={`${appOrigin}/on/demandware.store/Sites-${
                locals.site.id
            }-Site/${locals.locale.id.replace('-', '_')}/SLASSessionHelper-SyncHybridSession`}
```
If the `locals` values are not set, it seems like that should cause an error here, rather than a more confusing error later on when a bad request is made to `/on/demandware.store/Sites-undefined-Site/undefined/SLASSessionHelper-SyncHybridSession`.

If `locals` isn't set on first render, but _is_ set by the time we need the values:
```suggestion
            OCAPISessionsURL={locals.site?.id && locals.locale?.id && `${appOrigin}/on/demandware.store/Sites-${
                locals.site.id
            }-Site/${locals.locale.id?.replace('-', '_')}/SLASSessionHelper-SyncHybridSession`}
```
This way, we have the correct URL when `locals` is available, but still get a more helpful error if we try to `fetch(undefined)` before `locals` is set.

Reply:
Updated `OCAPISessionURL` to have conditional checks for `siteId` and `locale`, but we probably won't get a `fetch(undefined)` error. Before we make the fetch call to `OCAPISessionURL` in `packages/commerce-sdk-react/src/auth/index.ts` we check if `OCAPISessionURL` is undefined or not, so not having `siteId` and `locale` defined would result in no attempted fetch calls instead of an error